src/main/java/com/ibm/disni/util/MemoryAllocation.java

```diff
@@ -21,67 +21,155 @@
 
 package com.ibm.disni.util;
 
+import org.slf4j.Logger;
+
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
+import java.util.Comparator;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
 
-public class MemoryAllocation {
-	public enum MemType {
-		INDIRECT, DIRECT,
-	}
 
+public class MemoryAllocation {
```

Comment:
I believe the Memory Allocator is too "complicated". Given the MAX_CACHE_SIZE is already so low why would we need to shrink the cache in the first place? It is probably sufficient to just let the cache grow.